crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -45,9 +45,25 @@ def role_constraints
   # Unfortunatelly we need to explicitely look at crowbar-status of the proposal
   # because apply_role from this model ignores errors from superclass's apply_role.
   def commit_and_check_proposal
-    proposal_commit("default", false, false)
+    answer = proposal_commit("default", false, false)
+    # check if error message is saved in one of the nodes
+    if answer.first != 200
+      found_errors = []
+      NodeObject.find("state:crowbar_upgrade").each do |node|
+        error = node["crowbar_wall"]["chef_error"] || ""
+        next if error.empty?
+        found_errors.push error
+        @logger.error("Chef run ended with an error on the node #{name}: #{error}")
+        node["crowbar_wall"]["chef_error"] = ""
+        node.save
+      end
+      unless found_errors.empty?
+        raise found_errors.join("<br/>").html_safe
```

Comment:
I don't think you should use html here as this would look nice when the error appears in the json-based API.

There must be a way to have the `\n` bits translated to `<br/>` later on, but not sure how...